src/utils/googleTagManager.ts

```diff
@@ -155,14 +166,64 @@ export const trackEvent = ({
   action: string
   label?: EventLabel
 }): void => {
-  const dataLayer = {
+  const dataLayer: EventDataLayer = {
     event,
     chainId: _getChainId(),
     eventCategory: category,
     eventAction: action,
     eventLabel: tryParse(label),
   }
 
+  track(dataLayer)
+}
+
+type SafeAppEventDataLayer = {
+  event: GTM_EVENT.SAFE_APP
+  chainId: string
+  safeAppName: string
+  safeAppMethod: string
+  safeAppEthMethod?: string
+  safeAppSDKVersion?: string
+}
+
+export const getSafeAppName = (safeApp?: SafeApp): string => {
+  if (!safeApp?.id) {
+    return EMPTY_SAFE_APP
+  }
+
+  try {
+    const parsedSafeApp = JSON.parse(safeApp.id)
+
+    return parsedSafeApp.name || parsedSafeApp.url || parsedSafeApp
+  } catch (error) {
+    return EMPTY_SAFE_APP
+  }
```

Comment:
Can you not use `tryParse` from further up? You con't need to `try/catch`.
```suggestion
    const parsedSafeApp = tryParse(safeApp.id)
    return parsedSafeApp.name || parsedSafeApp.url || parsedSafeApp || EMPTY_SAFE_APP
```

Reply:
This is typed. I need to add SafeApp to the already numerous types and as well is intended to use with EventLabel not with SafeApp so is a little confusing